src/Symfony/Component/EventDispatcher/ContainerAwareEventDispatcher.php

```diff
@@ -105,7 +105,7 @@ public function removeListener($eventName, $listener)
     public function hasListeners($eventName = null)
     {
         if (null === $eventName) {
-            return (bool) count($this->listenerIds) || (bool) count($this->listeners);
+            return $this->listenerIds || $this->listeners;
```

Comment:
As somewhat not familiar with this piece of code, `return 0 < count($this->listenerIds) || 0 < count($this->listeners);` looks much easier to understand than `return (bool) count($this->listenerIds) || (bool) count($this->listeners)` and `return $this->listenerIds || $this->listeners;`. Performances are another subject.